### PR TITLE
Add HyperLogLog Data Type

### PR DESCRIPTION
**Pull request by alexmoore**
Adds HLL data type + bucket properties.

Test with a devrel of this Riak branch: https://github.com/basho/riak_ee/tree/develop-2.2

Docs PR forthcoming.

### CODE REVIEW

**Comment by alexmoore** [comment]
Docs at https://github.com/basho/private_basho_docs/commit/a126a2d2a5097e06b79062853e56769e3a346b2f.

File: lib/commands/crdt/fetchhll.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    return protobuf;
+
+};
+
+FetchHll.prototype.onSuccess = function(dtFetchResp) {
+    var cardinality = null;
+    var isNotFound = false;
+    // on a "not found" the value is null
+
+    if(dtFetchResp.value) {
```
Comment (lukebakken) [inline lib/commands/crdt/fetchhll.js]: LOL

Comment (alexmoore) [inline lib/commands/crdt/fetchhll.js]: :whups:

Comment (alexmoore) [inline lib/commands/crdt/fetchhll.js]: Should be "the cardinality is null"

File: test/unit/kv/fetchbuckettypeprops.js [hunk 2]
```diff
@@ -62,21 +62,22 @@ describe('FetchBucketTypeProps', function() {
             props.setBackend(new Buffer('backend'));
             props.setSearchIndex(new Buffer('index'));
             props.setDatatype(new Buffer('dataType'));
-            
+            props.setHllPrecision(14);
```
Comment (lukebakken) [inline test/unit/kv/fetchbuckettypeprops.js]: The tests for `HllPrecision` bucket / type properties should be in `test/integration/crdt`. Maybe rename the update/fetch test file to just `hll.js`.

If not, they'll fail forever until HLL support is in Riak.

Comment (alexmoore) [inline test/unit/kv/fetchbuckettypeprops.js]: I only added hllPrecision to the unit tests for the bucket / type props, so it shouldn't affect the existing integration tests. 

Also the new field is optional, so as long as the node PB library follow that it shouldn't choke riak up.      

```
// Hyperlolog DT Precision
    optional uint32 hll_precision = 29;
```

Comment (lukebakken) [inline test/unit/kv/fetchbuckettypeprops.js]: OK coolio

**Comment by lukebakken** [comment]
@alexmoore nice job :+1:

**Pull request merged**